### ISSUE TITLE
Update download shows (and makes) no progress for days

### ISSUE THREAD

**Issue by ascheman** [docker/for-mac#6001]
- [ ] I have tried with the latest version of Docker Desktop (-> **Thats what I am trying**)
  - [ ] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 08B19409-F57B-4704-A460-19BD1DEE2EDD/20211007045041

### Expected behavior

Available updates run smoothly and show a progress in some way (I would like to see how much of a download is already processed, ETA etc.).

### Actual behavior

Docker shows me there are updates available (for days now):

![image](https://user-images.githubusercontent.com/456000/136323976-677c2163-a660-490a-8169-149ed3b81937.png)

Clicking on the button shows a dialog stating that the update is in progress:

![image](https://user-images.githubusercontent.com/456000/136324080-cc0eb9f4-0795-4a52-afe0-d2cd3020fd19.png)

Nothing happens in the end. I have even restarted several times hoping that the download is already finished. And the respective button suggests the update might finish with a restart (but obviously it doesn't):

![image](https://user-images.githubusercontent.com/456000/136324363-9ddce83b-87fd-48d8-b388-52809255b803.png)

### Information

  - macOS Version: 11.6
  - Intel chip or Apple chip: 2,4 GHz 8-Core Intel Core i9
  - Docker Desktop Version:

![image](https://user-images.githubusercontent.com/456000/136324440-515b105a-5331-4e1a-8ab2-3060ed637012.png)

**easy copy to clipboard would be convenient**

### Steps to reproduce the behavior

See above

**Comment by ascheman** [comment]
Please find attached the output (I have removed some lines regarding BraveBrowser): 
[docker.log.gz](https://github.com/docker/for-mac/files/7319484/docker.log.gz)

**Comment by ascheman** [comment]
Today I magically managed to download and upgrade successfully to 4.1.0 ...

**Comment by ascheman** [comment]
Could you analyse the logs meanwhile, @trungutt?

**Comment by BIRDHEart** [comment]
I am seeing this issue as well.  Everyday when I log on I am asked to update but nothing seems to happen.

system dropdown says "Download will start soon..." 

I am also on 4.1.0

**Comment by Clashsoft** [comment]
Same problem on 4.2.0. I used `brew upgrade --cask --force docker` to upgrade from 4.1 → 4.2 already, but the issue is still here.

**Comment by Hyldegaard** [comment]
> Meanwhile I seem to have the problem again. The version 4.1.0 offers an update dialogue and a new update seems to be available: ![image](https://user-images.githubusercontent.com/456000/138033628-62b7fabe-bd34-4165-96b4-421879264ac2.png)
> 
> However, pressing the download button seems to lead to some background activity (button is greyed out): ![image](https://user-images.githubusercontent.com/456000/138033745-82dff7bc-2fec-4d29-a0c2-7ddfea365887.png)
> 
> But in the end nothing happens, after some hours I can press the button again ...

Same issue with 4.3.2 -> 4.4.2 on MacOS.

**Comment by ascheman** [comment]
Same here, having _Downloading update_ now for some days, it is really a mess!

![image](https://user-images.githubusercontent.com/456000/151708275-f723373b-097c-4f20-9a00-024924956416.png)

**Comment by berliner** [comment]
Same here with 4.3.2 -> 4.4.2 on MacOS

**Comment by morjfrog** [comment]
Same here with 4.3.2 -> 4.4.2 MacOS

**Comment by berliner** [comment]
Restarting Docker for Mac made the issue go away for me.

**Comment by morjfrog** [comment]
> Restarting Docker for Mac made the issue go away for me.

Yes for me as well. Still an annoying bug. No progress, and then a restart required.

**Comment by joshdurbin** [comment]
Same issue, I've re-installed Docker for Mac at this point at least 5 times to deal with this. The current stuck upgrade is `4.6.0` to `4.6.1`.

Docker says "A new Update is ready to download". The download button says "Downloading update" but it's been that way for days. The menu bar Docker widget thing shows "Download will start soon...".

**Comment by ascheman** [comment]
Meanwhile I installed docker-desktop via Homebrew and do frequently updates (`brew upgrade --cask`). This does not always run smoothly but anyways better than leaving the updates to the product itself.

**Comment by PillowPillow** [comment]
any ETA for this issue. Docker for mac stuck on downloading update for few weeks now.. (from 4.5.0 to 4.6.1)

**Comment by Yonben** [comment]
Same here, it says `You’re currently on version 4.5.0 (74594). The latest version is 4.7.1 (77678)`
But it's stuck on "Downloading update..." I use a MacBook Pro (14-inch, 2021) M1 Pro

**Comment by SiyiLi** [comment]
<img width="747" alt="image" src="https://user-images.githubusercontent.com/6840081/167524612-ab11b9a2-afd7-4674-a45e-1e689b4580c9.png">
Same here, downloading update forever

**Comment by rgarver** [comment]
Me too. 4.7.0 to 4.8.2. This is not the first time and I've had to do a fresh install most times.


<img width="1247" alt="Screen Shot 2022-05-24 at 10 43 55 PM" src="https://user-images.githubusercontent.com/6658/170188590-92dd7151-c49c-4c8c-8137-b0f54260e54a.png">

**Comment by md5orsha256** [comment]
I have the same problem. It also happens not for the first time.

<img width="1269" alt="image" src="https://user-images.githubusercontent.com/17884471/170440849-d2c7808f-599e-401c-9c54-d796c1087364.png">
<img width="265" alt="image" src="https://user-images.githubusercontent.com/17884471/170441196-8c6f613d-ffa4-4e3f-b99a-dac63cfdf0c9.png">

**Comment by tuxes3** [comment]
I had the same error for at least a month. As soon as I resized the Docker Disk Image (from 60GB to 120GB) it installed.

**Comment by max1c** [comment]
Same thing. Stuck on Downloading update macOS 4.9.1 -> 4.10.1.

**Comment by rweads0520** [comment]
@bmwilly comment worked for me. In case that link goes away: System Preferences / Security & Privacy / Privacy tab, select  Full Disk Access (click the lock to make changes), Add (+), browse to Applications and select Docker. Stop and restart docker desktop, update should complete download and be ready to restart and apply.

**Comment by nyetwurk** [comment]
Same problem
```You’re currently on version 4.10.1 (82475). The latest version is 4.12.0 (85629)```
Stuck on "Downloading update..."

**Comment by nyetwurk** [comment]
I think some sort of more useful error (e.g. "please give docker access to these folders" or "please give docker full disk access") would be helpful. There are also MacOS dev guidelines for Apps to automatically send users to the Privacy -> Disk Access tab.

**Comment by SethMilliken** [comment]
This is still a problem. The update experience in Docker is incredibly frustrating.

**Comment by zacharyfox** [comment]
Guess what? Still Broken!

**Comment by grantx2016** [comment]
Still broken for me also. I fail to understand why this is so difficult to resolve when every other macOS app updates seamlessly. 🤦

**Comment by minikdev** [comment]
let's keep this issue open! facing the same annoying problem at versions:

You’re currently on version 4.9.1 (81317). The latest version is 4.12.0 (85629)

**Comment by ryanrhee** [comment]
4.11.1 (84025). The latest version is 4.12.0 (85629)

Dashboard button says "Downloading update..." but menu bar icon menu says "Download will start soon..." ; maybe this discrepancy is helpful for debugging.

**Comment by dreadnautxbuddha** [comment]
@ryanrhee try removing the full disk access, then add it again. Docker already had full disk access on my machine, but removing the access, then adding it back again worked.

**Comment by ryanrhee** [comment]
removing and adding back disk access fixed it for me.

**Comment by brooklynb7** [comment]
> https://forums.docker.com/t/docker-desktop-wont-update-on-macos/114648/18 worked for me

This works! Thanks a lot!

**Comment by tim-arney** [comment]
Same problem on macOS 13.2.1 (22D68) with Apple M1 chip, trying to update Docker Desktop from 4.10.1 (82475) to 4.17.0 (99724)

**Comment by ndtreviv** [comment]
It downloads the update, enables the Install and Restart button, which shows a spinner, then quits Docker. No restart happens, and Activity Monitor shows all Docker processes are gone. When you start it up again there's no change - still a pending update that you have to install _all over again_.

**Comment by muescha** [comment]
> For anyone experiencing this, I resolved it be giving Docker Desktop app full disk access and file/folders access on the MAC Security privacy settings. After doing this, it successfully updated and gave a notification that Docker Desktop requires full permissions and you press OK and it updated correctly. Hope this helps.

Done - giving full disk access

but still the same - no update possible - no progress bar - traffic at Activity Monitor about 859MB - but the update button is not activated, just the update message disappear after some time :(

> `You're currently on version 4.22.0 (117440). The latest version is 4.23.0 (120376).`

**Comment by caarmen** [comment]
>but still the same - no update possible - no progress bar - traffic at Activity Monitor about 859MB - but the update button is not activated, just the update message disappear after some time :(

Just reading this gave me the idea to look at the Network tab in Activity Monitor, for the Docker process. I could at least see there that it was still actively downloading, and not stuck. 👍🏻 Thanks for the idea.

Of course would be better to have a progress bar in the Docker app 😅

**Comment by delisa-glover** [comment]
For 4.26.1 it was 408 MB in the Activity Monitor on the Network tab.

**Comment by douira** [comment]
It appears that it's only downloading the update at around 100kbit/s which is 100x slower than this internet connection would allow. Something is off
